Project/Source/Components/UI/ComponentCanvas.h

```diff
@@ -12,12 +12,15 @@ class ComponentCanvas : public Component {
 	void Update() override;
 	void DuplicateComponent(GameObject& owner) override;
 	void OnEditorUpdate() override;
+	bool CanBeRemoved() const override;
+
 	void SetScreenReferenceSize(float2 screenReferenceSize_);
 	void SetDirty(bool dirty_);	   //Dirty marks wether or not hte next time GetScreenFactor is called will be recalculated or not
 	float GetScreenFactor() const; //Returns the factor by which UI elements will be scaled (both x and y so it doesn't deform elements)
 
 private:
 	void RecalculateScreenFactor();
+	bool AnyChildHasCanvasRenderer(const GameObject*) const; //This method is only meant to be used to check if the canvas component can be destroyed
```

Comment:
En vez de especificar el propósito de la función, añadiría qué devuelve: "This method returns true if any child has a canvas renderer". Después de explicar que hace, adicionalmente puedes añadir el propósito